coreax/solvers/recombination.py

```diff
@@ -320,10 +340,14 @@ def _co_linearize(
     non_zero_weights_mask = weights > 0
     zero_weights_mask = 1 - non_zero_weights_mask
     n_zeros = zero_weights_mask.sum()
-    weights = weights.at[max_index].divide(n_zeros + 1)
+    # Create a new set of indices that replace the zero-weighted node indices with the
+    # maximum weighted node's index.
     indices = jnp.arange(weights.shape[0])
     indices *= non_zero_weights_mask
     indices += zero_weights_mask * max_index
+    # Renormalize the maximum weight; ensures the weight sum is preserved under the new
+    # (co-linearized) indices; prevents co-linearization from changing the weight sum.
+    weights = weights.at[max_index].divide(n_zeros + 1)
```

Comment:
`_resolve_null_basis` has some `Pyright` complaints that should be fairly simple to resolve

Reply:
Could you try fix the `Pyright` complaint at 405 too?

Reply:
These should be sorted in the latest commit.